### PR TITLE
Refactor layout of add-to-cart modal

### LINKED ISSUE #26642: [FO] Mobile add-to-cart modal is messed up on 1.7.8.x - regression
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

obraz

### Expected behavior

It was good enough on 1.7.7.x

### Steps to reproduce

1. Open product page on mobile or dev tools responsive mode
2. Add product to cart and wait for the modal to pop up

### PrestaShop version(s) where the bug happened

1.7.8.x

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by saulaski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | New responsive layout of add-to-cart modal with flex design. Checkout button is moved on top in mobile display.
| Type?             | improvement
| Category?         | FO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes #26642.
| How to test?      | <ol><li>Open product page on mobile or dev tools responsive mode</li><li>Add product to cart and wait for the modal to pop up</li></ol>
| Possible impacts? | Cart modal and navigation to order page.

# BC Break
The layout and style of the quickview model has changed, peoples using some selectors or overriding the style may have some issues

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27548)

### CODE REVIEW

**Comment by saulaski** [comment]
### Desktop layout
desktop-flex-layout
### Mobile layout
mobile-flex-layout

File: themes/classic/_dev/css/components/products.scss
Comment (NeOMakinG) [inline themes/classic/_dev/css/components/products.scss]:
```suggestion
      color: $gray-dark;
```

Please use SCSS variables everywhere when setting colors, the closest one, you can also play with `lighten` and `darken` when needed

File: themes/classic/modules/ps_shoppingcart/modal.tpl
Comment (NeOMakinG) [inline themes/classic/modules/ps_shoppingcart/modal.tpl]:
```suggestion
        <div class="cart-content-btn">
          <button type="button" class="btn btn-secondary" data-dismiss="modal">{l s='Continue shopping' d='Shop.Theme.Actions'}</button>
        </div>
        <div class="cart-content-btn checkout-action">
          <a href="{$cart_url}" class="btn btn-primary"><i class="material-icons rtl-no-flip">&#xE876;</i>{l s='Proceed to checkout' d='Shop.Theme.Actions'}</a>
        </div>
        <div class="card-content-footer">
          {hook h='displayCartModalFooter' product=$product}
        </div>
```

I'm pretty sure you'll be able to have the same result with a markup like this, I don't really understand the need of some line-break element

Comment (saulaski) [reply themes/classic/modules/ps_shoppingcart/modal.tpl]: I replaced the `break-line` markup with a recursive `flex-grid` behavior, where `flex-row` doesn't wrap content.
```
  .flex-grid {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-between;
    padding: 0;

    @include media-breakpoint-down(sm) {
      flex-flow: column nowrap;
    }

    & > * {
      padding: 1rem;

      @include media-breakpoint-down(sm) {
        flex-basis: 100%;
        padding: 0.5rem 0;
      }
    }

    & > .flex-grid {
      padding: 0;
    }
  }

  .flex-row {
    flex-basis: 100%;
    flex-wrap: nowrap;
  }
```

**Review by NeOMakinG — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by Progi1984** [comment]
@saulaski Needs a rebase

**Comment by saulaski** [comment]
Rebased ! Previous changes I made to fix version 1.7.8.x are now dismissed by the modal refactoring into flex layout.

**Review by NeOMakinG — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by NeOMakinG** [comment]
@prestascott we need you to unblock this PR :)

**Comment by prestascott** [comment]
Hi everyone,

@saulaski Please remove the check icon on the "Commander" button. 

Good to me for the desktop version.
Although, the "Commander" button needs to be placed just above the "Continuer mes achats" button, and not at the very top.

**Comment by saulaski** [comment]
Hello @prestascott, layouts have been updated according to your directions.

### Desktop Layout
desktop-flex-layout-2

### Mobile Layout
mobile-flex-layout-2

**Comment by prestascott** [comment]
Perfect, thanks! @saulaski 🎉

**Review by NeOMakinG — APPROVED** [review]

**Comment by saulaski** [comment]
Please wait before merging, because the new theme should include this layout instead of "classic" theme.

**Review by atomiix — APPROVED** [review]

**Comment by NeOMakinG** [comment]
> Please wait before merging, because the new theme should include this layout instead of "classic" theme.

Not both? The product team told you this? I think we didn't really work on the new theme add-to-cart modal yet so...

**Comment by saulaski** [comment]
> > Please wait before merging, because the new theme should include this layout instead of "classic" theme.
> 
> Not both? The product team told you this? I think we didn't really work on the new theme add-to-cart modal yet so...

We both heard the same decision collectively taken in stand up meeting a week ago: this refactoring should go into the new theme. I can't recall if product manager was present, I ask.
Please note that I wasn't delighted by this decision, but decisions are decisions.

**Comment by NeOMakinG** [comment]
> > > Please wait before merging, because the new theme should include this layout instead of "classic" theme.
> > 
> > 
> > Not both? The product team told you this? I think we didn't really work on the new theme add-to-cart modal yet so...
> 
> We both heard the same decision collectively taken in stand up meeting a week ago: this refactoring should go into the new theme. I can't recall if product manager was present, I ask. Please note that I wasn't delighted by this decision, but decisions are decisions.

It depends on one thing: Is the classic theme currently in maintenance only? If yes, then we should work on bugs only, not improvements, but this is something the product team should be aware of too!

**Comment by eternoendless** [comment]
Remember that themes from 1.7 must work with little or no changes in v8

**Comment by florine2623** [comment]
Hello all,

Thanks for the details !

I have tested this PR, LGTM QA ✅ .
I let you decide if you merge it or not. For now, it is blocked, we need Product team to approve it.

If another test is needed, please ping me and change QA label !

**Review by NeOMakinG — CHANGES_REQUESTED** [review]
Blocking it, waiting for a decision

**Comment by saulaski** [comment]
> Remember that themes from 1.7 must work with little or no changes in v8

The new layout is reusing the exact same CSS selectors. Of course HTML structure has changed a little:
- Bootstrap 3 Grid is layed aside, replaced with a custom Flex grid closer to Boostrap 4
- This refactoring allows more freedom about blocks & buttons placement in mobile display

@NeOMakinG even suggested to include this custom grid as a global alternative wherever Boostrap 3 Grid gets too limited.

@prestascott @MatShir what dou you guys think ? should we include this refactoring in version 8 ?

**Comment by matks** [comment]
@MatShir @fabienvallon this PR has been blocked for more than 12 days now 😭 can we speed things up?

**Comment by atomiix** [comment]
@MatShir @fabienvallon friendly reminder ;)

**Comment by MatShir** [comment]
Hi, 
We can't allow BC on the classic theme even for the v8.0, there are too many themes relying on it. 
Agreed with @matks, the refacto should be done on the new theme https://github.com/PrestaShop/hummingbird if you want to apply major improvement and features

**Comment by SharakPL** [comment]
@MatShir this modal was broken by 1.7.8 so it should be fixed here

**Comment by MatShir** [comment]
@SharakPL I believe, that the modal is fixed https://github.com/PrestaShop/PrestaShop/pull/27502. Here, it is an optimization/cleaning of the code, that can lead to BC for the classic theme.

**Comment by SharakPL** [comment]
I know it's been fixed already, but there still hasn't been new version of this module released since May last year and shops still use broken messed up version. What's the hold up?

**Comment by SharakPL** [comment]
Right. Sorry.

**Comment by matks** [comment]
Hello everyone looks like this PR got stale.

Going back through the notifications, I tried to summarize the constraints:

- PrestaShop 8.0.0 will be shipped with Classic theme as default theme as far as I know (because Hummingbird theme is not complete yet)
- Classic theme must be kept backward compatible as much as possible in order to ease the upgrade from PrestaShop 1.7 to PrestaShop 8
- The PHP backend code on PrestaShop 8 must be compatible with both Classic theme and Hummingbird theme in order to allow, when Hummingbird is complete, people to switch

Also when this PR was constructed the Classic theme was a part of this repository. The theme has now been moved to its own repository.

I think @saulaski won't work on this PR anymore so I close it for now. When someone picks up this topic again, he'll be able to reuse this branch content and possibly @saulaski 's commits (it's important to acknowledge his work) as a base and submit a new refactoring Pull Request against hummingbird theme only.

Thank you @saulaski and everyone involved in this PR.

**Pull request closed without merging**
